lib/ObjWriter/objwriter.cpp

```diff
@@ -622,6 +731,46 @@ extern "C" void EmitDebugFunctionInfo(ObjectWriter *OW,
   OST.EmitCVLinetableDirective(OW->FuncId++, Fn, FnEnd);
 }
 
+extern "C" void EmitDebugFunctionInfo(ObjectWriter *OW,
+                                      const char *FunctionName,
+                                      int FunctionSize) {
+  assert(OW && "ObjWriter is null");
+  auto *AsmPrinter = &OW->getAsmPrinter();
+  auto &OST = static_cast<MCObjectStreamer &>(*AsmPrinter->OutStreamer);
+  MCContext &OutContext = OST.getContext();
+  const MCObjectFileInfo *MOFI = OutContext.getObjectFileInfo();
+
+  if (MOFI->getObjectFileType() == MOFI->IsCOFF) {
+    EmitPDBDebugFunctionInfo(OW, FunctionName, FunctionSize);
+  } else {
+    // TODO: Should convert this for non-Windows.
+  }
+}
+
+extern "C" void EmitDebugVar(ObjectWriter *OW, char *name, int typeIndex,
+                             bool isParm, int rangeCount, char *ranges) {
+  assert(OW && "ObjWriter is null");
+
+  if (rangeCount == 0)
+    return;
+
+  // Interpret ranges as an array of NativeVarInfo structs.
+  ICorDebugInfo::NativeVarInfo *varInfos =
+      (ICorDebugInfo::NativeVarInfo *)ranges;
+
+  int varNumber = varInfos[0].varNumber;
+
+  DebugVarInfo var = {varNumber, name, typeIndex, isParm, NULL};
+  var.ranges = new std::vector<ICorDebugInfo::NativeVarInfo>();
+
+  for (int i = 0; i < rangeCount; i++) {
+    assert(var.varNumber == varInfos[i].varNumber);
```

Comment:
You could assert varNumber == varInfos[i].varNumber. But let's not allocate "varNumber" inside DebugVarInfo as mentioned above.